aptos-move/move-examples/post_mint_reveal_nft/sources/whitelist.move

```diff
@@ -0,0 +1,102 @@
+        let last_stage_end_time = vector::borrow(&whitelist_mint_config.whitelist_configs, vector::length(&whitelist_mint_config.whitelist_configs) - 1).whitelist_minting_end_time;
+        let now = timestamp::now_seconds();
+        now >= first_stage_start_time && now < last_stage_end_time
+    }
+
+    public(friend) fun get_num_of_stages(module_address: address): u64 acquires WhitelistMintConfig {
+        vector::length(&borrow_global<WhitelistMintConfig>(module_address).whitelist_configs)
+    }
+
+    public(friend) fun init_whitelist_config(admin: &signer) {
+        let config = WhitelistMintConfig {
+            whitelist_configs: vector::empty<WhitelistStage>(),
+        };
+        move_to(admin, config);
+    }
+
+    public(friend) fun add_whitelist_stage(admin: &signer, whitelist_start_time: u64, whitelist_end_time: u64, whitelist_price: u64, stage: u64) acquires WhitelistMintConfig {
+        assert!(whitelist_start_time < whitelist_end_time, error::invalid_argument(EINVALID_WHITELIST_SETTING));
+        let config = borrow_global_mut<WhitelistMintConfig>(signer::address_of(admin));
+        assert!(vector::length(&config.whitelist_configs) == stage, error::invalid_argument(EINVALID_STAGE));
+        let whitelist_stage = WhitelistStage {
+            whitelisted_address: bucket_table::new<address, u64>(4),
+            whitelist_mint_price: whitelist_price,
+            whitelist_minting_start_time: whitelist_start_time,
+            whitelist_minting_end_time: whitelist_end_time,
+        };
+        vector::push_back(&mut config.whitelist_configs, whitelist_stage);
+    }
+
+    public(friend) fun add_whitelist_addresses(admin: &signer, wl_addresses: vector<address>, mint_limit: u64, stage: u64) acquires WhitelistMintConfig {
+        let config = borrow_global_mut<WhitelistMintConfig>(signer::address_of(admin));
+        assert!(stage < vector::length(&config.whitelist_configs), error::invalid_argument(EINVALID_STAGE));
+        let whitelist_stage = vector::borrow_mut(&mut config.whitelist_configs, stage);
+        let now = timestamp::now_seconds();
+        assert!(now < whitelist_stage.whitelist_minting_end_time, error::invalid_argument(EINVALID_UPDATE_AFTER_MINTING));
+
+        let i = 0;
+        while (i < vector::length(&wl_addresses)) {
+            let addr = *vector::borrow(&wl_addresses, i);
+            // assert that the specified address exists
+            assert!(account::exists_at(addr), error::invalid_argument(EACCOUNT_DOES_NOT_EXIST));
+            bucket_table::add(&mut whitelist_stage.whitelisted_address, addr, mint_limit);
+            i = i + 1;
+        };
+    }
+
+    /// WhitelistMintConfigSingleStage stores information about one stage of whitelist.
+    public(friend) fun check_if_user_can_mint(module_address: address, minter_address: address, stage: u64, amount: u64): (u64, bool, bool) acquires WhitelistMintConfig {
```

Comment:
nit: Rename this to is_user_eligible_for_minting or can_user_mint. check_... implies it does a validation